tests/src/test/scala/spire/laws/cooperative_equality/Utility.scala

```diff
@@ -0,0 +1,76 @@
+package spire.laws.cooperative_equality
+
+import org.scalacheck.{Prop, Properties, Arbitrary, Gen}
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Prop._
+import org.scalacheck.Arbitrary._
+
+import spire.algebra.Semiring
+import spire.math.{SafeLong, Rational}
+import spire.implicits._
+
+object Utility {
+
+  def coopEquals[A](x: A, y: A): Boolean = 
+    x == y && x.hashCode == y.hashCode
+
+  val ALGEBRAIC  = "Algebraic"
+  val COMPLEX    = "Complex"
+  val JET        = "Jet"
+  val NATURAL    = "Natural"
+  val NUMBER     = "Number"
+  val QUATERNION = "Quaterion"
+  val REAL       = "Real"
+  val UBYTE      = "UByte"
+  val USHORT     = "UShort"
+  val UINT       = "UInt"
+  val ULONG      = "ULong"
+
+  val INT         = "Int"
+  val LONG 	      = "Long"
+  val DOUBLE      = "Double"
+  val FLOAT       = "Float"
+  val BIG_INT     = "BigInt"
+  val RATIONAL    = "Rational"
+  val BIG_DECIMAL = "BigDecimal"
+  val SAFE_LONG   = "SafeLong"
+  val STRING      = "String"
+  val BYTE        = "Byte"
+  val SHORT       = "Short"
+  val CHAR        = "Char"
+
+  val genSafeLong: Gen[SafeLong]                      = arbitrary[Long].map(SafeLong(_))
+  implicit val arbitrarySafeLong: Arbitrary[SafeLong] = Arbitrary(genSafeLong)
+
+  // Note - it's assumed that the SemiRing instance supports commutativity
+  // https://groups.google.com/forum/#!topic/spire-math/7r7SDbMd4AM
+  def props[A : Semiring](g: Gen[A]): List[Prop] = {
+    val commutativeAdd = forAll(g, g) { (a: A, b: A) =>
+      coopEquals(a+b,b+a) 
+    }
+    val commutativeMult = forAll(g, g) { (a: A, b: A) =>
+      coopEquals(a*b,b*a) 
+    }
+    val transitive = forAll(g, g, g) { (a: A, b: A, c: A) =>
+      if( coopEquals(a, b) && coopEquals(b, c)) coopEquals(a, c) else true
```

Comment:
Please let me know what you think of this property.

I added a print statement at one point when running this test. Only a handful of times did the if-statement evaluate to true.

But I wasn't sure of a better alternative...